securityscanutils/securityscan.go

```diff
@@ -233,10 +248,13 @@ func (r *SecurityScanRepo) RunMarkdownScan(ctx context.Context, release *github.
 			imageWithRepo = fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
 		}
 		fileName := fmt.Sprintf("%s_cve_report.docgen", image)
-		output := path.Join(outputDir, fileName)
+		output := path.Join(trivyScanOutputDir, fileName)
 		_, vulnFound, err := r.trivyScanner.ScanImage(ctx, imageWithRepo, markdownTplFile, output)
 		if err != nil {
-			return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+			if !errors.Is(err, ImageNotFoundError) {
```

Comment:
Will this ever be false? https://github.com/solo-io/go-utils/blob/main/securityscanutils/trivy_scanner.go#L91 we have some logic to handle missing errors, and we "swallow" the error.

Reply:
Ahh I see now when I looked lower at the code. I like the change? 

One thought - it seems we have two types of errors now: unrecoverableError (one that should cause us to short circuit), recoverableErr (one that we should continue the code execution). This would allow us to not require consumers of the `ScanImage` API, to be aware of the underlying reasons (that the image wasn't found), just whether or not the err should be recoverable or not.

not a requirement to implement, just figured I'd share the thought